IceCreamShop.playground/Contents.swift

```diff
@@ -1 +1,103 @@
+struct Flavor: Equatable {
+    let name: String
+    var rating: Double
+}
 
+enum Size: Double {
+    case small = 3.99
+    case medium = 4.99
+    case large = 5.99
+}
+
+struct Cone {
+    let flavor: Flavor
+    let topping: String?
+    let size: Size
+    
+    func eat() {
+        print("Mmm! \(flavor.name) is my favorite!")
+    }
+}
+
+class IceCreamShop {
+    var flavors: [Flavor]
+    var sizes: [Size]
+    var toppings: [String]
+    var totalSales: Double
+    
+    init(flavors: [Flavor], sizes: [Size], toppings: [String]) {
+        self.flavors = flavors
+        self.sizes = sizes
+        self.toppings = toppings
+        self.totalSales = 0.00
+    }
+    
+    func listTopFlavors() {
+        var topFlavors: [String] = []
+        var topFlavorString: String = "There are no flavors rated above 4.0."
+        for flavor in flavors {
+            if flavor.rating > 4.0 {
+                topFlavors.append(flavor.name)
+            }
+        }
+        for i in 0..<topFlavors.count {
+            if topFlavors.count == 1 {
+                topFlavorString = "Our top flavor is \(topFlavors[0])."
+            } else if i == 0 {
+                topFlavorString = "Our top flavors are \(topFlavors[i])"
+            } else if i != (topFlavors.count - 1) {
+                topFlavorString += ", \(topFlavors[i])"
+            } else {
+                topFlavorString += " and \(topFlavors[i])."
+            }
+        }
+        print(topFlavorString)
+    }
```

Comment:
You nailed this logic. It prints correctly no matter how many top flavors there are. Excellent job!